packages/spruce-skill-server/index.js

```diff
@@ -349,49 +364,51 @@ module.exports = async ({
 
 	// The logic before handle() is to suppress nextjs from responding and letting koa finish the request
 	// This allows our middleware to fire even after
-	router.get('*', async ctx => {
-		// if a controller already responded or we are making an API call, don't let next run at all
-		if (ctx.body || ctx.path.search('/api') === 0) {
-			debug('api call found, letting controllers handle it', ctx)
-			return
-		}
-		debug('handing off to next and backing off', ctx.path, ctx)
-		await handle(ctx.req, ctx.res)
-		ctx.respond = false
-		return
-
-		// this does not work as desired
-		ctx.body = await new Promise(resolve => {
-			const _end = ctx.res.end
-			ctx.res._end = _end
-
-			// Hijack stream to set ctx.body
-			const pipe = stream => {
-				ctx.res.end = _end
-				stream.unpipe(ctx.res)
-				resolve(stream)
+	if (!isApiOnly) {
+		router.get('*', async ctx => {
+			// if a controller already responded or we are making an API call, don't let next run at all
+			if (ctx.body || ctx.path.search('/api') === 0) {
+				debug('api call found, letting controllers handle it', ctx)
+				return
 			}
-			ctx.res.once('pipe', pipe)
-
-			// Monkey patch res.end to set ctx.body
-			ctx.res.end = body => {
-				debug('Next has finished for', ctx.path)
-				ctx.res.end = _end
-				ctx.res.removeListener('pipe', pipe)
-				if (ctx.res.redirect) {
-					debug('Next wants us to redirect to', ctx.res.redirect)
-					body = `Redirecting to ${ctx.res.redirect}`
-					ctx.redirect(ctx.res.redirect)
-					ctx.res.end(body)
-					// return
+			debug('handing off to next and backing off', ctx.path, ctx)
+			await handle(ctx.req, ctx.res)
+			ctx.respond = false
+			return
+
+			// this does not work as desired
```

Comment:
Is this still true? 😬 

I know this was existing code and you just reformatted but could you add more details?

Reply:
I didn't write this...not sure what this was referring to. @liquidg3 do you know?